src/Authentication/MsalServicePrincipalTokenProvider.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Identity.Client;
+
+namespace Microsoft.Artifacts.Authentication
+{
+    public class MsalServicePrincipalTokenProvider : ITokenProvider
```

Comment:
>MsalServicePrincipalTokenProvider [](http://example.com/codeflow?start=17&length=33)

consider updating MsalAuthenticationTests as well if it makes sense.. the idea there is it does a real token acquisition, so the tests are ignored by default, but can be run on-demand locally to ensure the token providers work

Reply:
I added tests and allowed client id and certificate to be passed with an env variable, though additional work to add a MI/SP to the feed